### DISCUSSION TITLE
Option to integrate with Traccar

### DISCUSSION THREAD

**Discussion by Tyler-2** [Xastir/Xastir#191] [Ideas]
I use Traccar as a self hosted open source device tracker. Clients, whether they be installed on phones or off the shelf GPS trackers, submit their location and other data (speed, heading, battery levels, engine ignition states, etc) at programmed intervals. As a result, I have a database of current and past GPS locations for all of my users' phones, which they submit via Wifi and LTE.

Xastir is an awesome package for coordination of location assets. Buuuut, most (almost all, really) of my tracked devices cannot submit data to APRS.fi, nor would I want them to due to the public nature of APRS.

If Xastir could pull info from a Traccar instance, you would be able to work with the location data of your LTE, Wifi, and RF/APRS devices all in one place.

I imagine this would just be another interface type configured in Interface Control.

Objects of interest to me are the devices themselves and any configured geofences.

Traccar's API info is here:
https://www.traccar.org/traccar-api/

**Comment by tvrusso** [comment]
I somehow missed this discussion when it was opened.  

There aren't any developers actively working on extending Xastir these days.  But what you're suggesting is similar to ADS-B and AIS (aircraft and ship) tracking information, which is also stuff that we don't want injected into APRS internet servers but which Xastir could display.

There are two scripts in the Xastir source tree (scripts/ads-b.pl and scripts/ais.pl) that read ADS-B and AIS data off of a software defined radio dongle (via third party software) and convert them into APRS item packets, which are then injected into Xastir through its UDP server port.

You could easily create something like that to access your tracar database through the API and construct a similar type of item packet for injection into Xastir.  By injecting it under a callsign/ssid other than the Xastir instance's own you could assure that Xastir doesn't adopt and retransmit the packet (either on RF or up to a connected APRS-IS server), and one wouldn't want to do this with an Xastir that is being run as an igate.

Perhaps those two scripts could serve as examples for you.
